File: LAB QUIZZES/CSE221_Quiz06/_712/_712.py
import os, sys, time, random, threading, subprocess
from _712_Solution import solve
start = 0; finish = 0; elapsed = 0
test = 0; best = 0; score = 0; batch = 0; total = 0
timeLimit = 1; nBatch = 10; N = 0; M = 0; S = 0; D = 0; K = 0; arr = []
adj = [[]]
u = []
v = []
w = []
weight = [ 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1]
nTest =  [0, 1, 50, 100, 100, 50, 20, 50, 50, 20, 50]
res = -1
edges = []
def printCase(verdict, test, N, M, S, D, K, edges):
    if os.path.getsize(verdict + ".txt") == 0:
        with open(verdict + ".txt", "w") as fout:
            fout.write("batch = " + str(batch) + "\ntest = " + str(test) + "\n")
            fout.write("N = " + str(N) + "\n")
            fout.write("M = " + str(M) + "\n")
            fout.write("S = " + str(S) + "\n")
            fout.write("D = " + str(D) + "\n")
            fout.write("K = " + str(K) + "\n")
            fout.write("edges = " + str(edges) + "\n")
    print(verdict + " on Batch " + str(batch), flush=True)
def exitCase(verdict):
    printCase(verdict, test, N, M, S, D, K, edges)
    os._exit(1)
def checkSoln():
    global start, finish, elapsed, N, M, S, D, K, arr, edges, res
    start = time.time()
    res = solve(N, M, S, D, K, edges)
    finish = time.time()
    elapsed += finish - start
    
    with open("_712_output.txt", "w") as fout:
        fout.write(str(res))
        
    ret = subprocess.run('_712_Judge.out _712_input.txt _712_output.txt', shell = True)
    if ret.returncode != 0:
        exitCase("WrongAnswer")

class MyThread(threading.Thread):
    def __init__(self, target, *args, **kwargs):
        super().__init__(target=target, args=args, kwargs=kwargs)
        self.exception = None

    def run(self):
        try:
            super().run()
        except Exception as e:
            self.exception = e

    def join(self, *args, **kwargs):
        super().join(*args, **kwargs)
        if self.exception:
            raise self.exception
        
def processBatch():
    global test, N, M, S, D, K, edges, u, v, w, arr
    # sys.setrecursionlimit(10**6)
    test = 1
    while test <= nTest[batch]:
        cmd = "_712_Gen.out " + str(batch) + " " + str(test) + " > _712_input.txt"
        subprocess.run(cmd, shell = True)
        with open('_712_input.txt') as temp:
            line1 = temp.readline().split()
            N = int(line1[0])
            M = int(line1[1])
            S = int(line1[2])
            D = int(line1[3])
            K = int(line1[4])
            u = list(map(int, temp.readline().split()))
            v = list(map(int, temp.readline().split()))
            w = list(map(int, temp.readline().split()))
            edges = []
            for i in range(M):
                edges.append((u[i], v[i], w[i]))

        threading.stack_size(128 * 1024 * 1024)  
        thread = MyThread(target=checkSoln)
        thread.start()
        try:
            thread.join()
        except Exception as e:
            print(e)
            exitCase("WrongAnswer")
        test += 1

def limitTime():
    recall = -1
    while elapsed < timeLimit and recall != elapsed:
        recall = elapsed
        time.sleep(timeLimit << 1)
    exitCase("TimeLimitExceeded")
if len(sys.argv) == 2:
    batch = int(sys.argv[1])
    random.seed(batch)
if 1 <= batch and batch <= nBatch:
    print("Running on Batch " + str(batch), flush=True)
    threading.Thread(target=limitTime, daemon=True).start()
    processBatch()
    print("Accepted on Batch " + str(batch) + " in " + f"{elapsed:.9f}" + "s", flush=True)
    os._exit(0)
ID = ""
with open("EnterIDandLanguage.txt") as fin:
    ID = fin.readline().split()[0]
if os.path.exists("_712_" + ID + ".py"):
    with open("_712_" + ID + ".py") as fin:
        best = int(fin.readline().split()[2])
open("WrongAnswer.txt", "w").close()
open("TimeLimitExceeded.txt", "w").close()
batch = 1
while batch <= nBatch:
    process = subprocess.run("pypy _712.py " + str(batch), capture_output=True, text=True)
    if process.stdout:
        print(process.stdout, end="")
    if process.stderr:
        print(process.stderr, end="")
    if process.returncode == 0:
        score += weight[batch]
    total += weight[batch]
    batch += 1
if best <= score:
    with open("_712_" + ID + ".py", "w") as fout:
        fout.write("# " + ID + " " + str(score) + "\n")
    subprocess.run("cmd /c echo # %COMPUTERNAME% %USERNAME%>>_712_" + ID + ".py")
    with open("_712_" + ID + ".py", "a") as fout:
        with open("_712_Solution.py") as code:
            fout.write(code.read())
print("Tentative score = " + str(score / total) + "/1", flush=True)
os._exit(0)

File: LAB QUIZZES/CSE221_Quiz06/_712/_712.java
import java.util.*;
import java.io.*;

public class _712 {
    public static Random rng = new Random();
    public static long start, finish, elapsed = 0, MOD = 1000003;
    public static int test = 0, best = 0, score = 0, batch = 0, total = 0;
    public static int timeLimit = 1, nBatch = 10;
    public static int weight[] = {  0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1};
    public static int nTest[] =  {0, 1, 50, 100, 100, 50, 20, 50, 50, 20, 50};
    public static int N, M, S, D, K;
    public static int res;
    public static ArrayList<Integer> arr;
    public static ArrayList<Integer> u, v, w;
    public static ArrayList<ArrayList<Integer> > edges;
    

    public static void printCase(String verdict, int test, int N, int M, int S, int D, int K, ArrayList <ArrayList<Integer> > edges)
            throws Exception {
        if (new File(verdict + ".txt").length() == 0) {
            BufferedWriter fout = new BufferedWriter(new FileWriter(verdict + ".txt"));
            fout.write("batch = " + batch + "\ntest = " + test + "\n");
            fout.write("N = " + N + "\n");
            fout.write("M = " + M + "\n");
            fout.write("S = " + S + "\n");
            fout.write("D = " + D + "\n");
            fout.write("K = " + K + "\n");
            fout.write("edges = " + edges + "\n");
            fout.close();
        }
        System.out.print(verdict + " on Batch " + batch + "\n");
    }

    public static void exitCase(String verdict) throws Exception {
        printCase(verdict, test, N, M, S, D, K, edges);
        System.exit(1);
    }

    public static void checkSoln() throws Exception {
        start = System.nanoTime();
        res = _712_Solution.solve(N, M, S, D, K, edges);
        finish = System.nanoTime();
        elapsed += finish - start;
        if (1000000000L * timeLimit < elapsed)
            exitCase("TimeLimitExceeded");

        BufferedWriter fout = new BufferedWriter(new FileWriter("_712_output.txt"));
        fout.write(Integer.toString(res) + "\n");
        fout.close();

        Process process = new ProcessBuilder("cmd.exe", "/c", "_712_Judge.out  _712_input.txt _712_output.txt").start();
        process.waitFor();
        if (process.exitValue() != 0)
            exitCase("WrongAnswer");
    }

    public static void processBatch() throws Exception {
        
        for (test = 1; test <= nTest[batch]; ++test) {
            String cmd = "_712_Gen.out " + batch + " " + test + " > _712_input.txt";
            Process process = new ProcessBuilder("cmd.exe", "/c", cmd).start();
            process.waitFor();

            BufferedReader fin = new BufferedReader(new FileReader("_712_input.txt"));
            String[] nums = fin.readLine().split(" ");
            N = Integer.parseInt(nums[0]);
            M = Integer.parseInt(nums[1]);
            S = Integer.parseInt(nums[2]);
            D = Integer.parseInt(nums[3]);
            K = Integer.parseInt(nums[4]);
            edges = new ArrayList<>();
            
            String[] temp = fin.readLine().split(" ");
            u = new ArrayList<>();
            for(int i = 0; i < M; i++){
                u.add(Integer.parseInt(temp[i]));
            }
            temp = fin.readLine().split(" ");
            v = new ArrayList<>();
            for(int i = 0; i < M; i++){
                v.add(Integer.parseInt(temp[i]));
            }
            temp = fin.readLine().split(" ");
            w = new ArrayList<>();
            for(int i = 0; i < M; i++){
                w.add(Integer.parseInt(temp[i]));
            }
            for(int i = 0; i < M; i++){
                ArrayList<Integer> ed = new ArrayList<>(Arrays.asList(u.get(i),v.get(i),w.get(i)));
                edges.add(ed);
            }

            checkSoln();
            fin.close();
        }
    }

    public static void limitTime() {
        try {
            for (long recall = -1; elapsed < 1000000000L * timeLimit && recall != elapsed;) {
                recall = elapsed;
                Thread.sleep(1000 * timeLimit << 1);
            }
            exitCase("TimeLimitExceeded");
        } catch (Exception e) {
            e.printStackTrace();
        }
    }

    public static void main(String args[]) throws Exception {
        if (args.length == 1)
            rng.setSeed(batch = Integer.parseInt(args[0]));
        if (1 <= batch && batch <= nBatch) {
            System.out.println("Running on Batch " + batch);
            new Thread(() -> limitTime()).start();
            processBatch();
            System.out.printf("Accepted on Batch " + batch + " in %.9fs\n", elapsed * 1e-9);
            System.exit(0);
        }
        BufferedReader fin = new BufferedReader(new FileReader("EnterIDandLanguage.txt"));
        String ID = fin.readLine().split("\\s+", 2)[0];
        fin.close();
        if (new File("_712_" + ID + ".java").exists()) {
            fin = new BufferedReader(new FileReader("_712_" + ID + ".java"));
            best = Integer.parseInt(fin.readLine().split("\\s+", 4)[2]);
            fin.close();
        }
        new FileOutputStream("WrongAnswer.txt").close();
        new FileOutputStream("TimeLimitExceeded.txt").close();
        for (batch = 1; batch <= nBatch; total += weight[batch], ++batch) {
            Process process = new ProcessBuilder("java", "-Xss128m", "_712", "" + batch).start();
            fin = new BufferedReader(new InputStreamReader(process.getInputStream()));
            fin.lines().forEach(System.out::println);
            fin.close();
            fin = new BufferedReader(new InputStreamReader(process.getErrorStream()));
            fin.lines().forEach(System.out::println);
            fin.close();
            if (process.waitFor() == 0)
                score += weight[batch];
        }
        if (best <= score) {
            BufferedWriter fout = new BufferedWriter(new FileWriter("_712_" + ID + ".java"));
            fout.write("// " + ID + " " + score + "\n");
            fout.close();
            new ProcessBuilder("cmd", "/c", "echo // %COMPUTERNAME% %USERNAME%>>_712_" + ID + ".java").start()
                    .waitFor();
            fout = new BufferedWriter(new FileWriter("_712_" + ID + ".java", true));
            fin = new BufferedReader(new FileReader("_712_Solution.java"));
            fout.write(fin.lines().collect(java.util.stream.Collectors.joining(System.lineSeparator())));
            fin.close();
            fout.close();
        }
        System.out.print("Tentative score = " + (score / (double) total) + "/1\n");
        System.exit(0);
    }
}
